src/librustc_mir/borrow_check/error_reporting.rs

```diff
@@ -445,30 +449,45 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
         scope_tree: &Lrc<ScopeTree>,
         borrow: &BorrowData<'tcx>,
         drop_span: Span,
-        borrow_span: Span,
-        _proper_span: Span,
+        borrow_spans: UseSpans,
         kind_place: Option<(WriteKind, &Place<'tcx>)>,
     ) -> DiagnosticBuilder<'cx> {
         debug!(
             "report_local_value_does_not_live_long_enough(\
              {:?}, {:?}, {:?}, {:?}, {:?}, {:?}\
              )",
-            context, name, scope_tree, borrow, drop_span, borrow_span
+            context, name, scope_tree, borrow, drop_span, borrow_spans
         );
 
-        let mut err = self.tcx.path_does_not_live_long_enough(
+        let borrow_span = borrow_spans.var_or_use();
+        let mut err = self.infcx.tcx.path_does_not_live_long_enough(
             borrow_span,
             &format!("`{}`", name),
             Origin::Mir,
         );
 
-        err.span_label(borrow_span, "borrowed value does not live long enough");
-        err.span_label(
-            drop_span,
-            format!("`{}` dropped here while still borrowed", name),
-        );
+        let explanation = self.explain_why_borrow_contains_point(context, borrow, kind_place);
+        if let Some(annotation) = self.annotate_argument_and_return_for_borrow(borrow) {
+            let region_name = annotation.emit(&mut err);
+
+            err.span_label(
+                borrow_span,
+                format!("`{}` would have to be valid for `{}`", name, region_name)
+            );
+            err.span_label(drop_span, format!("but `{}` dropped here while still borrowed", name));
```

Comment:
Given that this text always comes at the end and it's a continuation of a prior sentence, prepend `...` to signify this.

Reply:
Fixed this.